monix-eval/shared/src/main/scala/monix/eval/Task.scala

```diff
@@ -1234,6 +1234,12 @@ sealed abstract class Task[+A] extends Serializable {
   final def foreach(f: A => Unit)(implicit s: Scheduler): CancelableFuture[Unit] =
     foreachL(f).runAsync(s)
 
+  /** Returns a new `Task` that executes the source repetitively
+    * and never produces a value.
+    */
+  final def forever: Task[Nothing] =
+    flatMap(_ => this.forever)
```

Comment:
Is this stack safe ?

Reply:
Yes. Flat-mapping `Task`s is always safe, even in this case. This is also specified in `cats.effect.Sync`